test/test_cpp_api_parity.py

```diff
@@ -386,75 +418,129 @@ def test_methods(test_params):
         test_methods(test_params)
 
 
+def _compute_module_name(test_params_dict):
+    fullname = test_params_dict.get('fullname', None)
+    if fullname:
+        # NOTE: This doesn't work for some of the `wrap_functional` module tests such as "interpolate_nearest_1d",
+        # because in that case the module `interpolate` is not in `torch.nn` but rather in `torch.nn.functional`.
+        # We will fix this when we have parity tests for `torch.nn.functional` modules.
+        module_name = fullname.split('_')[0]
+    else:
+        module_name = test_params_dict.get('module_name')
+    return module_name
+
+
 def _process_test_params(test_params_dict, module_metadata, device):
-    module_name = test_params_dict.get('module_name')
+    module_name = _compute_module_name(test_params_dict)
     desc = test_params_dict.get('desc', None)
-    module_variant_name = module_name + (('_' + desc) if desc else '') + (('_' + device) if device != 'cpu' else '')
-    input_size = test_params_dict.get('input_size', None)
-    input_fn = test_params_dict.get('input_fn', None)
-    if input_size:
-        example_inputs = [torch.randn(input_size)]
-    elif input_fn:
-        example_inputs = list(input_fn())
+    module_variant_name = test_params_dict.get('fullname', module_name + (('_' + desc) if desc else ''))
```

Comment:
This looks like `module_variant_name = test.get_name()[5:] + (('_' + device) if device != 'cpu' else '')`, just to save some extra lines and using the logic inside TestBase.